From a8a54db08db2c1936642b10a787fd1754a3459e6 Mon Sep 17 00:00:00 2001
Message-Id: <a8a54db08db2c1936642b10a787fd1754a3459e6.1423872525.git.feitong.yi@intel.com>
In-Reply-To: <3062c7e6b919c393846ff0e13558e7f648b352ca.1423872525.git.feitong.yi@intel.com>
References: <3062c7e6b919c393846ff0e13558e7f648b352ca.1423872525.git.feitong.yi@intel.com>
From: Deepak M <m.deepak@intel.com>
Date: Mon, 26 Jan 2015 01:00:26 +0530
Subject: [PATCH 16/95] MUST_REBASE [VPG]: adf/intel: Adding PLL field in the
 attachment structure

What this patch does is:
- adds PLL filed in the intel_dc_attachment structure
- updates the pll_id values in attachments.

MUST_REBASE: Since Google ADF framework is not in upstream and also
we cannot have two display drivers, other one being i915, we need to
work on i915 and ADF convergence path before it can be upstreamed.

Issue: GMINL-5542
Change-Id: I75cf3368518cafd72b62848001276adabd002391
Signed-off-by: Deepak M <m.deepak@intel.com>
---
 drivers/video/adf/intel/core/vlv/vlv_dc_config.c   |   15 +++++++++++++++
 .../video/adf/intel/include/core/intel_dc_config.h |    1 +
 2 files changed, 16 insertions(+)

diff --git a/drivers/video/adf/intel/core/vlv/vlv_dc_config.c b/drivers/video/adf/intel/core/vlv/vlv_dc_config.c
index 5532833..59dbd38 100644
--- a/drivers/video/adf/intel/core/vlv/vlv_dc_config.c
+++ b/drivers/video/adf/intel/core/vlv/vlv_dc_config.c
@@ -25,38 +25,47 @@ static const struct intel_dc_attachment chv_allowed_attachments[] = {
 	{
 		.pipe_id = PIPE_A,
 		.plane_id = PRIMARY_PLANE,
+		.pll_id = PLL_A,
 	},
 	{
 		.pipe_id = PIPE_A,
 		.plane_id = SPRITE_A,
+		.pll_id = PLL_A,
 	},
 	{
 		.pipe_id = PIPE_A,
 		.plane_id = SPRITE_B,
+		.pll_id = PLL_A,
 	},
 	{
 		.pipe_id = PIPE_B,
 		.plane_id = SECONDARY_PLANE,
+		.pll_id = PLL_B,
 	},
 	{
 		.pipe_id = PIPE_B,
 		.plane_id = SPRITE_C,
+		.pll_id = PLL_B,
 	},
 	{
 		.pipe_id = PIPE_B,
 		.plane_id = SPRITE_D,
+		.pll_id = PLL_B,
 	},
 	{
 		.pipe_id = PIPE_C,
 		.plane_id = TERTIARY_PLANE,
+		.pll_id = PLL_C,
 	},
 	{
 		.pipe_id = PIPE_C,
 		.plane_id = SPRITE_E,
+		.pll_id = PLL_C,
 	},
 	{
 		.pipe_id = PIPE_C,
 		.plane_id = SPRITE_F,
+		.pll_id = PLL_C,
 	}
 };
 
@@ -64,26 +73,32 @@ static const struct intel_dc_attachment vlv_allowed_attachments[] = {
 	{
 		.pipe_id = PIPE_A,
 		.plane_id = PRIMARY_PLANE,
+		.pll_id = PLL_A,
 	},
 	{
 		.pipe_id = PIPE_A,
 		.plane_id = SPRITE_A,
+		.pll_id = PLL_A,
 	},
 	{
 		.pipe_id = PIPE_A,
 		.plane_id = SPRITE_B,
+		.pll_id = PLL_A,
 	},
 	{
 		.pipe_id = PIPE_B,
 		.plane_id = SECONDARY_PLANE,
+		.pll_id = PLL_B,
 	},
 	{
 		.pipe_id = PIPE_B,
 		.plane_id = SPRITE_C,
+		.pll_id = PLL_B,
 	},
 	{
 		.pipe_id = PIPE_B,
 		.plane_id = SPRITE_D,
+		.pll_id = PLL_B,
 	}
 };
 
diff --git a/drivers/video/adf/intel/include/core/intel_dc_config.h b/drivers/video/adf/intel/include/core/intel_dc_config.h
index 2183c4c..b8fbbcd 100644
--- a/drivers/video/adf/intel/include/core/intel_dc_config.h
+++ b/drivers/video/adf/intel/include/core/intel_dc_config.h
@@ -440,6 +440,7 @@ struct intel_pipe {
 struct intel_dc_attachment {
 	u8 plane_id;
 	u8 pipe_id;
+	u8 pll_id;
 };
 
 struct intel_pipeline {
-- 
1.7.9.5

